### ISSUE TITLE
Free first person camera

### ISSUE THREAD

**Issue by DuendeInexistente** [simlu/voxelshop#249]
I get this is intended primarily at designing small things, not actual voxel maps.
But would it be hard to add a free first person camer? One that can do more than rotating around the central point? It'd make my life so much easier.

**Comment by simlu** [comment]
Not hard, but I have limited time atm. Do you know that you can change the central point that the camera moves around? Does that work for you?

Otherwise can you specify your use case a bit more please?

**Comment by DuendeInexistente** [comment]
Oh, I can change the central point? I haven't found the option to do that. I did find the hotkeys to pan the camera up/down/left/right thought, and those were pretty useful.
My use case is mainly wanting to make small maps, similar to models or very, very broad sketches of an area.

**Comment by simlu** [comment]
Yes, please take a look at the documentation here:
https://simlu.github.io/voxelshop/voxelshop_shortcuts.html
 
Should help you until we get to adding a free camera in!

**Comment by JohnPablok** [comment]
Hello! Is there any way I can change how the "middle mouse button" and the "right mouse button" move the camera? I want to rotate the view using right click, and pan the view using middle button, which is the of how it is right now. I don't see any options for that on the Shortcut manager.

**Comment by simlu** [comment]
There is currently no way to change thar unfortunately.

But basically you want the same camera shortcuts that blender has im guessing? I think that's on the too somewhere.

**Comment by JohnPablok** [comment]
Yes exactly! That's the setup I use on Blender, it's also very similar to the one in Unity and even Maya (omitting the fact that you have to press Alt there). The switched functionality messes a lot with my muscle memory, I guess it happens to other people too.

**Comment by simlu** [comment]
Let me take a look

**Comment by JohnPablok** [comment]
Thank you! By the way, apparently my memory sucks. In Unity you can pan with middle mouse button and rotate the camera with right click, all without pressing ALT. Pressing ALT actually rotates with LBM, pans with MMB and zooms with RMB. Maya might be the same.

**Comment by simlu** [comment]
I did some work here towards customizing mouse / camera actions. Will try to do a QA release shortly...

https://github.com/simlu/voxelshop/pull/250

**Comment by JohnPablok** [comment]
Wow! Awesome! Thank you so much, I'll be sure to test it as soon as it's released.

**Comment by simlu** [comment]
Ok, release is up: https://github.com/simlu/voxelshop/releases

**Comment by JohnPablok** [comment]
Nice! It works beautifully! The only problem I notice is that it isn't possible to rotate the camera using RMB while I'm using the "Draw voxel" tool and the mouse is in the voxel zone, of course this is by design because right click erases voxels, but it might be troublesome in big scenes, doesn't matter though, I can live with that haha.
 I really want to thank you, this is going to improve my workflow a lot!

**Comment by simlu** [comment]
Doesn't `ctrl` work to always enable the camera?

**Comment by JohnPablok** [comment]
Oooh, well that solves it! Awesome, yes, it works.
